packages/at_client/lib/src/util/at_client_validation.dart

```diff
@@ -97,14 +97,15 @@ class AtClientValidation {
     }
     // If key starts with 'cached:' (or) if the metadata of key has isCached set to true; it represent
     // a cached key. Prevent client updating the cached key. Hence throw exception.
-    if (atKey.key!.startsWith('$CACHED:') ||
+    if (atKey.key!.startsWith('${AtConstants.cached}:') ||
         (atKey.metadata != null && atKey.metadata!.isCached)) {
       throw AtKeyException('Cannot create/update a cached key');
     }
     // If namespace is not set on key and in preferences, throw exception
     if ((atKey.namespace == null || atKey.namespace!.isEmpty) &&
```

Comment:
change this condition to
if (((atKey.namespace == null || atKey.namespace!.isEmpty &&) !atKey.isLocal) &&
        (atClientPreference.namespace == null ||
            atClientPreference.namespace!.isEmpty)
        )

Reply:
change made as part of [27600db](https://github.com/atsign-foundation/at_client_sdk/pull/1135/commits/27600dbb07a1f422a7e254200b1007e97f54567d)